//
// Copyright 2021-2022 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
syntax = "proto2";

package magritte;

import "mediapipe/framework/calculator.proto";

message RotationMode {
  enum Mode {
    UNKNOWN = 0;
    ROTATION_0 = 1;
    ROTATION_90 = 2;
    ROTATION_180 = 3;
    ROTATION_270 = 4;
  }
}

message RotationCalculatorOptions {
  extend mediapipe.CalculatorOptions {
    optional RotationCalculatorOptions ext = 355176967;
  }
  optional RotationMode.Mode rotation_mode = 1;
  optional bool clockwise = 2 [default = false];
}
